doc/src/modules/physics/mechanics/joints.rst

```diff
@@ -17,52 +17,68 @@ Joints in Physics/Mechanics
-and ``child_joint_pos``. The orientation of the joint in each body is defined by
-the ``parent_axis`` and ``child_axis``. These two vectors are aligned as
-explained in the :class:`~.Joint` notes and are in the image parallel to the
-red vector. As last the joint also needs :func:`~.dynamicsymbols` as generalized
-coordinates and speeds. In the case of the :class:`~.PrismaticJoint` shown
-above, the generalized coordinate ``q_1`` distance along the joint axis.
-And the generalized speed ``u_1`` is its velocity.
+in green) and the child body (shown in blue). The attachment of the joint to the
+body consists of a point and a frame. In the parent body the point is called
+``parent_point`` and the frame ``parent_interframe``. For the child body these
+are called ``child_point`` and ``child_interframe``. The arguments for of the
+attachment points are ``parent_joint_pos`` and ``child_joint_pos``.
+
+For describing the joint transformation the joint generally needs
+:func:`~.dynamicsymbols` for the generalized coordinates and speeds. Some joints
+like the :class:`~.PinJoint`, :class:`~.PrismaticJoint` also require a
+``joint_axis``, which is an axis that does not change between the
+``parent_interframe`` and ``child_interframe``. So in case of the
+:class:`~.PinJoint`, also shown below, this means that the ``joint_axis`` is the
+axis of rotation. With the generalized coordinate :math:`\theta` as the angle of
+rotation and the generalized speed :math:`\omega` as the angular velocity.
+
+.. image:: api/PinJoint.svg
+   :align: center
+   :width: 600
 
 With the information listed above, the joint defines the following
 relationships. It first defines the kinematic differential equations, which
 relate the generalized coordinates to the generalized speeds. Next, it orients
 the parent and child body with respect to each other. After which it also
 defines their velocity relationships.
 
-The following code shows the creation of a :class:`~.PrismaticJoint` as shown
-above with arbitrary linked position vectors: ::
+The code below shows the creation of a :class:`~.PinJoint` as shown above
+with arbitrary linked position vectors. In this code the attachment points are
+set using vectors, which define the attachment point with respect to the body's
+mass center. The intermediate frames are not set, so those are the same as the
+body's frame. ::
 
    >>> from sympy.physics.mechanics import *
    >>> mechanics_printing(pretty_print=False)
-   >>> q1, u1 = dynamicsymbols('q1, u1')
+   >>> theta, omega = dynamicsymbols('theta, omega')
    >>> parent = Body('parent')
    >>> child = Body('child')
-   >>> joint = PrismaticJoint(
-   ...     'slider', parent, child, q1, u1,
-   ...     parent_joint_pos=parent.frame.x / 2 + parent.frame.y / 10,
-   ...     child_joint_pos=-(child.frame.x + child.frame.y) / 10,
-   ...     parent_axis=parent.frame.x, child_axis=child.frame.x)
+   >>> joint = PinJoint(
+   ...     'hinge', parent, child, theta, omega,
```

Comment:
Can we make `theta` and `omega` keyword arguments here? Otherwise it relies on the person reading the documentation having good understanding on `PinJoint`s constructor signature.